virtwho/config.py

```diff
@@ -1264,6 +1288,117 @@ def _validate(self):
         self.validation_messages.extend(validation_messages)
 
 
+class LibvirtdConfigSection(VirtConfigSection):
+    """
+    This class is used for validation of libvirtd virtualization backend
+    section. It tries to validate options and combination of options that
+    are specific for this virtualization backend.
+    """
+
+    VIRT_TYPE = 'libvirt'
+
+    def __init__(self, section_name, wrapper):
+        super(LibvirtdConfigSection, self).__init__(section_name, wrapper)
+
+    def _validate_server(self):
```

Comment:
I would suggest that we do whatever sort of checking is necessary for the scheme etc being used throughout libvirtd here. (See Libvirtd._get_url)

In this way we can show what sort of problems might occur long before we try to use the values.

Reply:
I want to clarify some things.

Your suggestion seems logical, but I don't see any validation in `Libvirtd._get_url()` in terms of exit this method unless condition is met. What I see there is logging of warnings, when scheme is not complete and some default value is injected to scheme.

I don't recommend moving content of  `Libvirtd._get_url()` to `LIbvirtdConfigSection._validate_server()`, because purpose of all `_validate*()` method is validation of configuration files/CLI/ENV and not changing of loaded configuration options, right?

If we want to change loaded options in `ConfigSection` and subclasses of `ConfigSection`, then I'm not against, but we should do it in another method.

Reply:
The purpose of _validate*() methods is validating the config values from whatever source they may be as well as to make any changes necessary for that parameter to be usable (so long as there is a reasonable way to do that with the given parameter).

There are more things I'd like to be able to verify (like the connectivity to the source/destinations for the given configuration) but such things can wait.